simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -294,6 +347,58 @@ KOKKOS_FORCEINLINE_FUNCTION simd<T, simd_abi::scalar> condition(
                                                         : static_cast<T>(c));
 }
 
+template <class T, class BinaryOperation>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
+reduce(Experimental::simd<T, Experimental::simd_abi::scalar> const& x,
+       Experimental::simd_mask<T, Experimental::simd_abi::scalar> const& mask,
+       T identity, BinaryOperation) noexcept {
+  if (!mask) return identity;
+  return x[0];
+}
+
+template <class T, class BinaryOperation>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
+reduce(Experimental::simd<T, Experimental::simd_abi::scalar> const& x,
+       BinaryOperation binary_op) noexcept {
+  return reduce(
+      x, Experimental::simd<T, Experimental::simd_abi::scalar>::mask_type(true),
+      T(0), binary_op);
+}
```

Comment:
Is this overload necessary? It seems dangerous to implicitly assume `0` to be the neutral element.

Reply:
The overload is necessary as this is part of the reduction interface in std::simd.
As for the identity element, it does look like `0` would not be a correct value to put in according to std::simd:
```
The default argument for identity_element is equal to
  • T() if BinaryOperation is plus<>,
  • T(1) if BinaryOperation is multiplies<>,
  • T(~T()) if BinaryOperation is bit_and<>,
  • T() if BinaryOperation is bit_or<>, or
  • T() if BinaryOperation is bit_xor<>.
```
I'll get it fixed.